pkg/controller/linstorsatelliteset/linstorsatelliteset_controller.go

```diff
@@ -178,22 +196,24 @@ func (r *ReconcileLinstorSatelliteSet) Reconcile(request reconcile.Request) (rec
 	}
 
 	if err := r.addFinalizer(ctx, pns); err != nil {
-		return reconcile.Result{}, err
+		log.WithField("err", err).Warn("failed to add finalizer to resource")
+		errs = append(errs, err)
 	}
 
 	// Create the satellite configuration
 	configMap, err := reconcileSatelliteConfiguration(ctx, pns, r)
 	if err != nil {
-		return reconcile.Result{}, err
```

Comment:
I looks like a a nil `configMap` will also cause problems. Generally I'm a bit uncomfortable with this continue on errors approach. Even functions that don't return a value might have side-effects which are necessary for later steps to work. Could we restructure this function like this?

```
Get(pns)
err := doStuff(pns)
reconcileStatus(pns, err)
```

So that `doStuff` can return at any point and we still update the status?

If we really do perform independent steps here, then we could also build that into the code structure somehow to make it clearer what requires what.

Reply:
So I reworked the SatelliteSet controller to follow this approach. 

Apart from cosmetic changes (renamed every satelliteset instance to `satelliteSet` instead of `nodeSet`, `pns`, etc...)
I also removed the linstor client from the reconciler object. This makes it easy to ensure that its initialized at usage site.